### PR TITLE
aws-crt-cpp: add version 0.17.12 with update dependencies

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **aws-crt-cpp/0.17.12**

One of the PRs to solve https://github.com/conan-io/conan-center-index/issues/7763.
Updates almost dependent recipes.

On the other hand, since aws-crt-cpp/ 0.17.11, there are a compilation error caused by latest compiler 
I made quick patch for it. I will make PR to original repository later.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/aws-crt-cpp/all/conanfile.py
Comment (SpaceIm) [inline recipes/aws-crt-cpp/all/conanfile.py:87]: no set_property() for the moment please, or use absolute target (conan >= 1.43.0).

Comment (toge) [reply recipes/aws-crt-cpp/all/conanfile.py:87]: OK, I will revert.

Comment (toge) [reply recipes/aws-crt-cpp/all/conanfile.py]: Fixed

**Review by danimtb — APPROVED** [review]

File: recipes/aws-crt-cpp/all/conanfile.py
Comment (SpaceIm) [inline recipes/aws-crt-cpp/all/conanfile.py:4]:
```suggestion
required_conan_version = ">=1.33.0"
```

File: recipes/aws-crt-cpp/all/conanfile.py
Comment (SpaceIm) [inline recipes/aws-crt-cpp/all/conanfile.py:24]: is is a new option? I mean, did it exist in 0.14.3? Why openssl is not in requirements()?

File: recipes/aws-crt-cpp/all/conanfile.py
Comment (SpaceIm) [inline recipes/aws-crt-cpp/all/conanfile.py:19]:
```suggestion
```

File: recipes/aws-crt-cpp/all/conanfile.py
Comment (SpaceIm) [inline recipes/aws-crt-cpp/all/conanfile.py:65]:
```suggestion
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]

File: recipes/aws-crt-cpp/all/conanfile.py
Comment (SpaceIm) [inline recipes/aws-crt-cpp/all/conanfile.py:13]: can be replaced by:

```python
    def export_sources(self):
        self.copy("CMakeLists.txt")
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
            self.copy(patch["patch_file"])
```

It allows to not export unrelated patches.

**Review by SSE4 — APPROVED** [review]

File: recipes/aws-crt-cpp/all/test_package/CMakeLists.txt
Comment (SSE4) [inline recipes/aws-crt-cpp/all/test_package/CMakeLists.txt:5]: good catch!

**Comment by SpaceIm** [comment]
Don't approve yet, there is a new option we don't want, and it's always tedious to remove options.

**Comment by ericLemanissier** [comment]
@toge could you please push a change (even broken) to invalid the existing reviews ? some reviewers (me included) sometimes approve PRs without checking the conversation.

**Review by uilianries — CHANGES_REQUESTED** [review]
Blocking to avoid the merge. Please, apply SpaceIm suggestion.

File: recipes/aws-crt-cpp/all/conanfile.py
Comment (toge) [reply recipes/aws-crt-cpp/all/conanfile.py:13]: OK. I will replace yours.

**Comment by toge** [comment]
@SpaceIm @ericLemanissier  @uilianries 
Sorry for late.
If I will be in same situation, I will push a change or turn PR draft. 
Thank you for your help.

File: recipes/aws-crt-cpp/all/conanfile.py
Comment (SpaceIm) [inline recipes/aws-crt-cpp/all/conanfile.py:59]: try to keep execution order, it's quite misleading to see export_sources() after source(), actually it's completly unrelated.

**Comment by SpaceIm** [comment]
By the way, now you can use set_property() and bump conan min version to 1.43 if you want. conan version of CCI has been updated.

File: recipes/aws-crt-cpp/all/conanfile.py
Comment (toge) [reply recipes/aws-crt-cpp/all/conanfile.py:59]: Sorry, I forget this documents.
https://github.com/conan-io/conan-center-index/blob/master/docs/reviewing.md#order-of-methods-and-attributes

I will fix it.

**Comment by toge** [comment]
@SpaceIm 
I modified this recipe to use `set_property()`.
I'm glad you to review it.

**Review by SpaceIm — APPROVED** [review]

**Review by ericLemanissier — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Comment by ericLemanissier** [comment]
ok, so I think this is stuck until @uilianries approves it

**Review by uilianries — APPROVED** [review]
LGTM

**Comment by MartinDelille** [comment]
I can't build aws-crt-cpp locally when running:
```
$ conan create aws-crt-cpp/all aws-crt-cpp/0.17.12@ --profile Darwin.txt
...
-- Library aws-c-s3 found /Users/martin/.conan/data/aws-c-s3/0.1.29/_/_/package/79abc8431f1c7df1253aabbe8290eeec551cd640/lib/libaws-c-s3.a
-- Found: /Users/martin/.conan/data/aws-c-s3/0.1.29/_/_/package/79abc8431f1c7df1253aabbe8290eeec551cd640/lib/libaws-c-s3.a
-- Configuring done
CMake Error at source_subfolder/CMakeLists.txt:270 (add_library):
  Target "aws-crt-cpp" links to target "AWS::aws-c-event-stream" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?


-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DATAROOTDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR


CMake Generate step failed.  Build files cannot be regenerated correctly.
aws-crt-cpp/0.17.12:
aws-crt-cpp/0.17.12: ERROR: Package '7855eb1fcc148ee1fb4c24b70d0f7c0f7551c835' build failed
aws-crt-cpp/0.17.12: WARN: Build folder /Users/martin/.conan/data/aws-crt-cpp/0.17.12/_/_/build/7855eb1fcc148ee1fb4c24b70d0f7c0f7551c835
ERROR: aws-crt-cpp/0.17.12: Error in build() method, line 73
	cmake = self._configure_cmake()
while calling '_configure_cmake', line 67
	self._cmake.configure()
	ConanException: Error 1 while executing cd '/Users/martin/.conan/data/aws-crt-cpp/0.17.12/_/_/build/7855eb1fcc148ee1fb4c24b70d0f7c0f7551c835' && cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE="Release" -DCMAKE_OSX_ARCHITECTURES="x86_64" -DCMAKE_OSX_SYSROOT="/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk" -DCONAN_IN_LOCAL_CACHE="ON" -DCONAN_COMPILER="apple-clang" -DCONAN_COMPILER_VERSION="13.0" -DCONAN_CXX_FLAGS="-m64" -DCONAN_SHARED_LINKER_FLAGS="-m64" -DCONAN_C_FLAGS="-m64" -DCONAN_LIBCXX="libc++" -DBUILD_SHARED_LIBS="OFF" -DCMAKE_INSTALL_PREFIX="/Users/martin/.conan/data/aws-crt-cpp/0.17.12/_/_/package/7855eb1fcc148ee1fb4c24b70d0f7c0f7551c835" -DCMAKE_INSTALL_BINDIR="bin" -DCMAKE_INSTALL_SBINDIR="bin" -DCMAKE_INSTALL_LIBEXECDIR="bin" -DCMAKE_INSTALL_LIBDIR="lib" -DCMAKE_INSTALL_INCLUDEDIR="include" -DCMAKE_INSTALL_OLDINCLUDEDIR="include" -DCMAKE_INSTALL_DATAROOTDIR="share" -DCONAN_CMAKE_POSITION_INDEPENDENT_CODE="ON" -DCMAKE_MODULE_PATH="/Users/martin/.conan/data/aws-crt-cpp/0.17.12/_/_/build/7855eb1fcc148ee1fb4c24b70d0f7c0f7551c835" -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY="ON" -DCONAN_EXPORTED="1" -DBUILD_TESTING="False" -DBUILD_DEPS="False" -Wno-dev '/Users/martin/.conan/data/aws-crt-cpp/0.17.12/_/_/build/7855eb1fcc148ee1fb4c24b70d0f7c0f7551c835'
make: *** [create] Error 1
```

Here is my profile:

```
[settings]
arch=x86_64
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=13.0
os=Macos
[options]
rtmidi:shared=False
```

**Comment by MartinDelille** [comment]
Ok I fix this by building aws-c-event-stream locally.

**Pull request merged**
